src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareReaderFailoverHandler.java

```diff
@@ -343,16 +346,19 @@ private ConnectionAttemptResult getNextResult(CompletionService<ConnectionAttemp
     private final ConnectionProvider connProvider;
```

Comment:
should we make this class non-static so that we don't have to pass in so many parameters to the constructor? Might be cleaner

Reply:
Sure, I'd keep just newHostTuple as a single constructor parameter then.